deploy/role_binding.yaml

```diff
@@ -2,7 +2,6 @@ kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: jaeger-operator
-  namespace: observability
```

Comment:
If the namespace in the role_binding from the deploy directory is being removed, it should be removed from the other files in the directory as well (operator.yaml, service_account.yaml). The readme has to be changed as well, to include `-n observability` to the `kubectl` commands.

cc @objectiser : I think this came from a change of yours. Do you remember why you needed this?

Reply:
`ClusteRole` and `ClusterRoleBindings` are cluster-wide objects the namespace is ignored. This patch does not change the way how the operator is being installed.